Lib/fractions.py

```diff
@@ -247,12 +248,23 @@ def limit_denominator(self, max_denominator=1000000):
             n, d = d, n-a*d
 
         k = (max_denominator-q0)//q1
-        bound1 = Fraction(p0+k*p1, q0+k*q1)
-        bound2 = Fraction(p1, q1)
-        if abs(bound2 - self) <= abs(bound1-self):
-            return bound2
+        bound1_n = p0 + k*p1
+        bound1_d = q0 + k*q1
+        bound2_n = p1
+        bound2_d = q1
+        bound1_minus_self_n = abs((bound1_n * d_original)
+                                  - (n_original * bound1_d))
+        bound1_minus_self_d = d_original * bound1_d
+        bound2_minus_self_n = abs((bound2_n * d_original)
+                                  - (n_original * bound2_d))
+        bound2_minus_self_d = d_original * bound2_d
+
+        difference = ((bound1_minus_self_n * bound2_minus_self_d)
+                      - (bound2_minus_self_n * bound1_minus_self_d))
+        if difference >= 0:
```

Comment:
Suggest a more direct `if diff1_n * diff2_d >= diff2_n * diff1_d: ...`. (It also saves an operation.)

Reply:
Thanks @mdickinson  -- it's late here (Hawai'i) so will get this first thing in the morning.  Thanks for the close review.

Reply:
Yep, there's no hurry here.